dashboard/app/models/user_level.rb

```diff
@@ -49,19 +49,23 @@ def readonly_requires_submitted
   end
 
   def best?
-    Activity.best? best_result
+    return false if best_result.nil?
+    best_result == ActivityConstants::BEST_PASS_RESULT
   end
 
   def perfect?
-    Activity.perfect? best_result
+    return false if best_result.nil?
+    best_result > ActivityConstants::MAXIMUM_NONOPTIMAL_RESULT
   end
 
   def finished?
-    Activity.finished? best_result
+    return false if best_result.nil?
+    best_result >= ActivityConstants::MINIMUM_FINISHED_RESULT
   end
 
   def passing?
-    Activity.passing? best_result
+    return false if best_result.nil?
+    best_result >= ActivityConstants::MINIMUM_PASS_RESULT
```

Comment:
Presumably these constants will at some point need to be renamed. Any reason for that point to not be now?